Add Spotify Player Widget

anant-j opened the pull request:
Hi,
With release of Glance v0.8.0, an external server is no longer required for Spotify's Now Playing Widget.
The token retrieval and then queue retrieval calls are made subsequently within this widget.

This PR eliminates the requirement to run an external service for this widget.

This also resolves issue #63  
Thanks

anant-j commented:
Hi @NeedsAdjustment, can you please review this PR.
Thanks

NeedsAdjustment commented:
Hey - awesome to see that Glance now has the functionality to handle the server-side calls, and the usage here looks great. I personally don't prefer the queue implementation and visual/textual tweaks in this PR, as I mentioned previously in #63 - I'm not fully opposed to a queue, but I think there are more minimal ways to do it (e.g. a single or double line of text below the main now-playing, rather than rendering the full cover art and song/artist names without any changes to the visual hierarchy).

I understand you're keen to build this out into a full player, so I'm wondering if just adding a new widget to the repo would be more appropriate considering the scope of the changes?

anant-j commented:
@NeedsAdjustment these are very valid points.
As suggested, I have moved this to a new widget and reverted the changes made to your widget.
Thanks for the feedback!!

svilenmarkov commented on widgets/spotify-player-by-anant-j/README.md:
```diff
@@ -0,0 +1,118 @@
+          {{ if and $isDeviceActive (not $isPrivateSession) }}
+            {{ if $isCurrentlyPlaying }}
+              <div style="margin-right:10px;">
+                <button style="font-size:25px;" onclick="fetch('https://api.spotify.com/v1/me/player/pause',{method:'PUT',headers:{'Authorization':'Bearer {{$accessToken}}'}}); setTimeout(function(){ location.reload(); }, 2000);">⏸</button>
```
While this works, I think there should be a gigantic warning that the user's access token will be exposed to the front-end and what the security implications of that are.

anant-j replied:
Agreed. Added warning on top of the readme.
Thanks

anant-j commented:
Hi @svilenmarkov, can you please review again?
Thanks

svilenmarkov reviewed (approved):
Thanks for adding the warning and for contributing!
